cmd/otel-allocator/internal/collector/collector.go

```diff
@@ -133,3 +142,25 @@ func (k *Watcher) runOnCollectors(store cache.Store, fn func(collectors map[stri
 func (k *Watcher) Close() {
 	close(k.close)
 }
+
+func isPodUnhealthy(pod *v1.Pod, collectorNotReadyGracePeriod time.Duration) bool {
+	if collectorNotReadyGracePeriod == config.DefaultCollectorNotReadyGracePeriod {
```

Comment:
Suggestion: check that `collectorNotReadyGracePeriod` is equal to 0 here rather than relying on the default value. In the future we may want to change the default in which case we'd need to change this condition.